src/CMakeLists.txt

```diff
@@ -796,6 +800,10 @@ target_link_libraries(ceph-dencoder
   ${EXTRALIBS}
   ${CMAKE_DL_LIBS}
   )
+if(${WITH_KVS})
```

Comment:
please squash this change into a dedicated commit which includes the https://github.com/ceph/ceph/pull/10024/commits/60185795b9c13fba8ae7947ba95ce7c18eed79c1#diff-95e351a3805a1dafa85bf20b81d086e6L747 changes. that change only handles the `WITH_KVS` related changes on FreeBSD.